### PR TITLE
fix: should infer package manager from config if there's no lockfile in the project

### PR DESCRIPTION
**Pull request by sodatea**
Fixed monorepo use cases.

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Underlying tools
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

**Other information:**

### CODE REVIEW

File: packages/@vue/cli/lib/util/ProjectPackageManager.js [hunk 1]
```diff
@@ -88,8 +89,17 @@ class PackageManager {
     if (forcePackageManager) {
```
Comment (Akryum) [inline packages/@vue/cli/lib/util/ProjectPackageManager.js]: How do we force the package manager?

File: packages/@vue/cli-shared-utils/lib/env.js [hunk 2]
```diff
@@ -130,6 +130,21 @@ function checkPnpm (result) {
...
+const _npmProjects = new LRU({
+  max: 10,
+  maxAge: 1000
+})
+exports.hasProjectNpm = (cwd) => {
+  if (_npmProjects.has(cwd)) {
+    return true
+  }
+
+  const lockFile = path.join(cwd, 'package-lock.json')
```
Comment (Akryum) [inline packages/@vue/cli-shared-utils/lib/env.js]: Maybe we should use findup?

File: packages/@vue/cli/lib/util/ProjectPackageManager.js (hunk 1 above)
Comment (sodatea) [reply packages/@vue/cli/lib/util/ProjectPackageManager.js]: The `-m / --packageManager` option of `vue create`. Maybe we should add this option for `vue add` and `vue invoke` too?

File: packages/@vue/cli-shared-utils/lib/env.js (hunk 2 above)
Comment (sodatea) [reply packages/@vue/cli-shared-utils/lib/env.js]: Yeah that makes more sense.

**Pull request merged**